packages/mui-joy/src/Modal/useModal.ts

```diff
@@ -0,0 +1,214 @@
+'use client';
+import * as React from 'react';
+import {
+  unstable_ownerDocument as ownerDocument,
+  unstable_useForkRef as useForkRef,
+  unstable_useEventCallback as useEventCallback,
+} from '@mui/utils';
+import { ModalManager, ModalOwnProps } from '@mui/base/Modal';
+import { EventHandlers, extractEventHandlers } from '@mui/base/utils';
+
+export interface UseModalRootSlotOwnProps {
+  role: React.AriaRole;
+  onKeyDown: React.KeyboardEventHandler;
+  ref: React.RefCallback<Element> | null;
+}
+
+export interface UseModalBackdropSlotOwnProps {
+  'aria-hidden': React.AriaAttributes['aria-hidden'];
+  onClick: React.MouseEventHandler;
+  open?: boolean;
+}
+
+export type UseModalBackdropSlotProps<TOther = {}> = TOther & UseModalBackdropSlotOwnProps;
+
+export type UseModalRootSlotProps<TOther = {}> = TOther & UseModalRootSlotOwnProps;
+
+type UseModalParameters = Pick<
+  ModalOwnProps,
+  'container' | 'disableEscapeKeyDown' | 'disableScrollLock' | 'open'
+> & {
+  'aria-hidden'?: React.AriaAttributes['aria-hidden'];
+  onClose?: {
+    bivarianceHack(event: {}, reason: 'backdropClick' | 'escapeKeyDown' | 'closeClick'): void;
+  }['bivarianceHack'];
+  onKeyDown?: React.KeyboardEventHandler;
+  ref: React.Ref<HTMLElement>;
+};
+
+function ariaHidden(element: Element, show: boolean): void {
```

Comment:
Shouldn't `ariaHidden` be imported from Base UI?
```suggestion
```

Reply:
Yep, I missed that it is implemented in Base UI too.